app/frontend/wallet/cardano-ledger-crypto-provider.js

```diff
@@ -0,0 +1,124 @@
+const TransportU2F = require('@ledgerhq/hw-transport-u2f').default // for browser
+const Ledger = require('ledgerhq/hw-app-ada').default // temporary hack, should be @ledgerhq
+const CachedDeriveXpubFactory = require('./helpers/CachedDeriveXpubFactory')
+const cbor = require('borc')
+const {TxWitness, SignedTransactionStructured} = require('./transaction')
+
+const CardanoLedgerCryptoProvider = async (ADALITE_CONFIG, walletState) => {
+  const transport = await TransportU2F.create()
+  transport.setExchangeTimeout(15 * 60000) // 60000ms = 1minute
+  const ledger = new Ledger(transport)
+
+  const state = Object.assign(walletState, {
```

Comment:
is this `Object.assign` intentional? Shouldn't this be using `setState` ?

Reply:
It was inspired by trezor crypto provider, but i don't understand why it should by using setState. It seems like constant read-only state used only by this class.

Reply:
at cardano wallet level, there is no "setState" function for changing the state, this is used only on the preact layer. This state isn't the global app state, it's just the state of the cardano wallet provider.

Reply:
Then why dont' we manipulate walletState directly?

Reply:
Yes, we can, it would be more straightforward. I'm not sure at this moment why we used Object.assign, it was probably a half-hearted effort to make a "local" state, but it's probably quite misleading, since this "local" state has also access to the global one.

Reply:
But probably I would not go for refactoring it now, and keep the pattern for the sake of consistency - I created a separate issue for that: https://github.com/vacuumlabs/adalite/issues/383